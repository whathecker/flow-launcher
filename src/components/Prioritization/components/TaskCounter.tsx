import React from "react";
import { Text } from "../../Themed";
import { Typography, Color } from "../../../styles";

type TaskCounterProps = {
  currentTaskIndex: number;
  totalTasksLength: number;
};

const TaskCounter: React.FC<TaskCounterProps> = ({
  currentTaskIndex,
  totalTasksLength,
}: TaskCounterProps) => {
  const currentTaskCounter = currentTaskIndex + 1;
  return (
    <>
      <Text
        style={{
          ...Typography.p,
          fontSize: 14,
          color: Color.light.subtleLabel,
        }}
      >{`Task (${currentTaskCounter} of ${totalTasksLength})`}</Text>
    </>
  );
};

export default TaskCounter;
