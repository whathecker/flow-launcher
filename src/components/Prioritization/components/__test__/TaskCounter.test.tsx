import * as React from "react";
import renderer from "react-test-renderer";
import TaskCounter from "../TaskCounter";

describe("Test TaskCounter component", () => {
  it("Should renders TaskCounter correctly", () => {
    const tree = renderer
      .create(<TaskCounter currentTaskIndex={2} totalTasksLength={5} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
