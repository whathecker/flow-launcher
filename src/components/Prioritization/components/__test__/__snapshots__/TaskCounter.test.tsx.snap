// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TaskCounter component Should renders TaskCounter correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
      },
      Object {
        "color": "#848484",
        "fontFamily": "Lexend-Regular",
        "fontSize": 14,
      },
    ]
  }
>
  Task (3 of 5)
</Text>
`;
